### ISSUE TITLE
NOTIFICATION_TRANSFORM_CHANGED cannot be consistently blocked with set_notify_transform(false)

### ISSUE THREAD

**Issue by Phischermen** [godotengine/godot#37283]
Godot 3.2

Windows

Description:
I was hoping to make a gdscript plugin that allowed me to arrange spatial nodes in a specific grid pattern. I wanted to use transform notifications to do this, so I added a node called "SpatialGridItem" which I parented under the spatial node I wanted to snap to the grid. I used set_notify_transform(false) to temporarily disable transform notifications so I could apply a translation with out starting an endless notification cycle.

```
func _notification(what):
        match what:
        NOTIFICATION_TRANSFORM_CHANGED:
            set_notify_transform(false)
            #snap parent to grid
            set_notify_transform(true)

```
It worked great, but I wanted to display a gizmo so I changed the base type for SpatialGridItem from Spatial to 3DPosition. My plugin suddenly got endless transform notifications the moment the transform was changed (this was the specific issue I tried to avoid). I experimented by changing SpatialGridItem's type to various different types, and I'm struggling to find a correlation!

Types that cause the endless notifications:
- Mesh Instance
- 3DPosition
- Camera

Types that don't cause endless notifications
- Spatial
- Navigation
- ImmediateGeometry

Minimal Reproduction:
[TransformNotificationIssue.zip](https://github.com/godotengine/godot/files/4378146/TransformNotificationIssue.zip)

**Comment by Phischermen** [comment]
After spending awhile trying to fix this, I have just found out that the function I need to get the results I want already exists! It's called set_ignore_transform_notification(bool). I changed my code to use this function instead, and now it doesn't send endless transform notifications.
```
func _notification(what):
        match what:
        NOTIFICATION_TRANSFORM_CHANGED:
            set_ignore_transform_notification(true)
            translation = Vector3.ZERO
            set_ignore_transform_notification(false)
```
That said, gizmos can look awkward while being translated, and if the user enters values directly into the inspector, gizmos will not wind up where they're supposed to go. For instance, if I translated a Camera node right by 5 units, and the script example above was attached, the final translation would be (0,0,0) but the gizmo would be located (5,0,0).

I think my new goal is to get the gizmos to follow their nodes no matter what.
